Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -472,6 +472,33 @@ public void DeleteItem(BaseItem item, DeleteOptions options, BaseItem parent, bo
             ReportItemRemoved(item, parent);
         }
 
+        private bool IsInternalItem(BaseItem item)
+        {
+            if (!item.IsFileProtocol)
+            {
+                return false;
+            }
+
+            var pathToCheck = item switch
+            {
+                Genre => _configurationManager.ApplicationPaths.GenrePath,
+                MusicArtist => _configurationManager.ApplicationPaths.ArtistsPath,
+                MusicGenre => _configurationManager.ApplicationPaths.GenrePath,
+                Person => _configurationManager.ApplicationPaths.PeoplePath,
+                Studio => _configurationManager.ApplicationPaths.StudioPath,
+                Year => _configurationManager.ApplicationPaths.YearPath,
+                _ => null
+            };
+
+            var itemPath = item.Path;
+            if (!string.IsNullOrEmpty(pathToCheck) && !string.IsNullOrEmpty(itemPath))
+            {
+                return itemPath.StartsWith(pathToCheck, StringComparison.OrdinalIgnoreCase);
```

Comment:
I don't know if I would trust this comparison. I believe we already have this implemented somewhat correctly somewhere. Unix is also case sensitive.